### PR TITLE
soloCup

### PR DESCRIPTION
**Pull request by PhantomEpicness**
## About The Pull Request

Plastic cups that can be crushed

Screenshot_1

## Why It's Good For The Game

New disposable cup yay. Requested by thwomper

## Changelog

:cl: TotalEpicness
add: Added new plastic cups! Credit to purplecloud for the sprites.
/:cl:

### CODE REVIEW

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 1]
```diff
@@ -281,6 +281,37 @@
...
+		to_chat(user,SPAN_WARNING("You crush \the [src]!"))
+		visible_message(SPAN_WARNING("[user] crushes the cup!"))
+		if(reagents.total_volume > 0)
+			reagents.clear_reagents()
+			playsound(src.loc, 'sound/effects/slosh.ogg', 25, 1, 3)
+			to_chat(user,SPAN_WARNING("\The contents of \the [src] spill!"))
+		new /obj/item/crushed_cup(user.loc)
+		qdel(src)
+
+/obj/item/crushed_cup
```
Comment (Segrain) [inline code/game/objects/items/reagent_containers/food/drinks.dm:306]:
```suggestion
		new /obj/item/trash/crushed_cup(user.loc)
		qdel(src)

/obj/item/trash/crushed_cup
```
Can also move crushed one to `trash.dm`.

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 2]
```diff
@@ -281,6 +281,37 @@
...
+	throwforce = 0
+	w_class = SIZE_TINY
+	matter = list("plastic" = 5)
+	attack_verb = list("bludgeoned", "whacked","slapped")
+
+/obj/item/reagent_container/food/drinks/cup/attack_self(mob/user)
+	. = ..()
+	if(user.a_intent == INTENT_HARM)
+		to_chat(user,SPAN_WARNING("You crush \the [src]!"))
+		visible_message(SPAN_WARNING("[user] crushes the cup!"))
```
Comment (Segrain) [inline code/game/objects/items/reagent_containers/food/drinks.dm:298]:
```suggestion
		user.visible_message(SPAN_WARNING("[user] crushes \the [src]!"), SPAN_WARNING("You crush \the [src]!"))
```

**Review by Segrain — CHANGES_REQUESTED** [review]

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 3]
```diff
@@ -281,6 +281,37 @@
...
+		if(reagents.total_volume > 0)
+			reagents.clear_reagents()
+			playsound(src.loc, 'sound/effects/slosh.ogg', 25, 1, 3)
+			to_chat(user,SPAN_WARNING("\The contents of \the [src] spill!"))
```
Comment (stanalbatross) [inline code/game/objects/items/reagent_containers/food/drinks.dm:302]:
```suggestion
			to_chat(user,SPAN_WARNING("The contents of \the [src] spill!"))
```
you do not need a backslash unless it's before a thing

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by c4xmaniac** [comment]
I love you, red solo cup, I lift you up, proceed to party, proceed to party. I’ve seen you in blue and I’ve seen you in yellow, but only you red will do for this fellow. You are the abbot to my Costello, and I’m hoping to get - lucky. Red solo cup, I lift you up

**Comment by MusselMain** [comment]
Can't wait to get me a red Solo cup full of death juice so I can get the lethal sip after FOB falls and I'm abandoned at the office

**Comment by PhantomEpicness** [comment]
ready

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 4]
```diff
@@ -281,6 +281,36 @@
 	else
 		icon_state = "water_cup_e"
 
+/obj/item/reagent_container/food/drinks/cup
+	name = "Plastic Cup"
```
Comment (Geevies) [inline code/game/objects/items/reagent_containers/food/drinks.dm:285]:
```suggestion
	name = "plastic cup"
```

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 5]
```diff
@@ -281,6 +281,36 @@
...
+	desc = "A generic red cup. Beer pong, anyone?"
+	icon = 'icons/obj/items/cup.dmi'
+	icon_state = "solocup"
+	throwforce = 0
+	w_class = SIZE_TINY
+	matter = list("plastic" = 5)
+	attack_verb = list("bludgeoned", "whacked","slapped")
```
Comment (Geevies) [inline code/game/objects/items/reagent_containers/food/drinks.dm:292]:
```suggestion
	attack_verb = list("bludgeoned", "whacked", "slapped")
```

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 6]
```diff
@@ -281,6 +281,36 @@
...
+
+/obj/item/reagent_container/food/drinks/cup/attack_self(mob/user)
+	. = ..()
+	if(user.a_intent == INTENT_HARM)
+		user.visible_message(SPAN_WARNING("[user] crushes \the [src]!"), SPAN_WARNING("You crush \the [src]!"))
+		if(reagents.total_volume > 0)
+			reagents.clear_reagents()
+			playsound(src.loc, 'sound/effects/slosh.ogg', 25, 1, 3)
+			to_chat(user,SPAN_WARNING("The contents of \the [src] spill!"))
```
Comment (Geevies) [inline code/game/objects/items/reagent_containers/food/drinks.dm:301]:
```suggestion
			to_chat(user, SPAN_WARNING("The contents of \the [src] spill!"))
```

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 7]
```diff
@@ -281,6 +281,36 @@
...
+		new /obj/item/trash/crushed_cup(user.loc)
+		qdel(src)
+
+/obj/item/trash/crushed_cup
+	name = "Crushed cup"
```
Comment (Geevies) [inline code/game/objects/items/reagent_containers/food/drinks.dm:306]:
```suggestion
	name = "crushed cup"
```

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 8]
```diff
@@ -281,6 +281,36 @@
...
+	desc = "A sad crushed and destroyed cup. It's now useless trash. What a waste."
+	icon = 'icons/obj/items/cup.dmi'
+	icon_state = "crushed_solocup"
+	throwforce = 0
+	w_class = SIZE_TINY
+	matter = list("plastic" = 5)
+	attack_verb = list("bludgeoned", "whacked","slaped")
```
Comment (Geevies) [inline code/game/objects/items/reagent_containers/food/drinks.dm:313]:
```suggestion
	attack_verb = list("bludgeoned", "whacked", "slapped")
```

**Review by Geevies — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
@Geevies please remember to draft when you request changes

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 9]
```diff
@@ -281,6 +281,36 @@
...
+
+/obj/item/reagent_container/food/drinks/cup/attack_self(mob/user)
+	. = ..()
+	if(user.a_intent == INTENT_HARM)
+		user.visible_message(SPAN_WARNING("[user] crushes \the [src]!"), SPAN_WARNING("You crush \the [src]!"))
+		if(reagents.total_volume > 0)
+			reagents.clear_reagents()
+			playsound(src.loc, 'sound/effects/slosh.ogg', 25, 1, 3)
+			to_chat(user, SPAN_WARNING("The contents of \the [src] spill!"))
+		new /obj/item/trash/crushed_cup(user.loc)
```
Comment (stanalbatross) [inline code/game/objects/items/reagent_containers/food/drinks.dm:302]: should this not go into your hand?

**Review by Agameofscones — COMMENTED** [review]
sprites LGTM

File: code/game/objects/items/reagent_containers/food/drinks.dm [hunk 10]
```diff
@@ -281,6 +281,36 @@
...
+/obj/item/reagent_container/food/drinks/cup/attack_self(mob/user)
+	. = ..()
+	if(user.a_intent == INTENT_HARM)
+		user.visible_message(SPAN_WARNING("[user] crushes \the [src]!"), SPAN_WARNING("You crush \the [src]!"))
+		if(reagents.total_volume > 0)
+			reagents.clear_reagents()
+			playsound(src.loc, 'sound/effects/slosh.ogg', 25, 1, 3)
+			to_chat(user, SPAN_WARNING("The contents of \the [src] spill!"))
+		new /obj/item/trash/crushed_cup(user.hand)
+		qdel(src)
```
Comment (stanalbatross) [inline code/game/objects/items/reagent_containers/food/drinks.dm:303]:
```suggestion
		qdel(src)
		var/obj/item/trash/crushed_cup/C = new /obj/item/trash/crushed_cup(user)
		user.equip_to_slot_if_possible(C, (user.hand ? WEAR_L_HAND : WEAR_R_HAND))
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**